### DISCUSSION TITLE
Suggester : list based on files in a folder

### DISCUSSION THREAD

**Discussion by davidszerman** [SilentVoid13/Templater#982] [Help] [answered]
Hello, 

I knew templater but I am currently going in the details of this great plugin.

The suggester feature is great

What I was able to do so fare is 
<% tp.system.suggester(["A", "B"], ["[[A]]", "[[B]]"]) %>

What I would like : to have a folder "Answers" and inside it several notes A, B,...
And the suggester look at this folder in order to get the list of values.

Is it possible ?

Thanks

**Comment by Zachatoo** [comment, marked as answer]
Definitely possible! We can use the Obsidian API to do this. We use `app.vault.getMarkdownFiles()` to get all the markdown files in the vault, then `.filter()` them down to only files in a specific folder. Then use `tp.system.suggester` using those markdown files.

```js
<%*
const items = app.vault.getMarkdownFiles()
	.filter(x => x.parent?.path === "Answers");
const selectedItem = (await tp.system.suggester((item) => item.basename, items)).basename;
-%>

[[<% selectedItem %>]]
```

Also linked answer [here](https://zachyoung.dev/snippets/templater#suggester-for-files-in-a-specific-folder) among other answers.

**Reply by Zachatoo** [reply]
Alternatively, could be more similar to your answer with the links as part of the selectedItem.

```js
<%*
const items = app.vault.getMarkdownFiles()
	.filter(x => x.parent?.path === "Answers");
const selectedItem = (await tp.system.suggester((item) => item.basename, items)).basename;
const link = "[[" + selectedItem + "]]";
-%>

<% link %>
```
